[BUG]: getFontPathByWeight fails when font has multiple entries per weight with different formats

**Describe the bug**

When changing the default font (e.g., from "Google Sans Code" to "Inter"), OG image generation fails during `npm run build` with `Error: Cannot find the font path`.

The root cause is in `src/utils/getFontPathByWeight.ts`. When `fontData` contains multiple entries for the same weight but with different format sets (e.g., one entry with only `woff2`, another with `woff` + `truetype`), the function uses `Array.find()` which stops at the first weight match — even if that entry doesn't contain the requested format. The inner `.find()` then returns undefined, and the function returns undefined.

**To Reproduce**

1. Replace the font configuration in `astro.config.ts` to use Inter instead of Google Sans Code:

```typescript
fonts: [
  {
    name: "Inter",
    cssVariable: "--font-inter",
    provider: fontProviders.google(),
    weights: [400, 500, 600, 700],
    styles: ["normal"],
    formats: ["woff2", "woff", "ttf"],
  },
],
```

2. Update references to `--font-google-sans-code` in `src/pages/og.png.ts` and `src/pages/posts/[...slug]/index.png.ts` to `--font-inter`.

3. Update the font family name in the satori JSX from "Google Sans Code" to "Inter".

4. Run `npm run build`.

5. Observe the build fails with `Error: Cannot find the font path`.

**Expected behavior**

The build should complete successfully and OG images should be generated using the new font.

**Screenshots**

Build error output:

```
[ERROR] Error: Cannot find the font path.
    at Module.GET (file:///opt/build/repo/dist/.prerender/chunks/og_XXXXX.mjs:12:11)
    at renderEndpoint (file:///opt/build/repo/dist/.prerender/chunks/prerender_XXXXX.mjs:7152:32)
    at PagesHandler.handle (file:///opt/build/repo/dist/.prerender/chunks/prerender_XXXXX.mjs:13216:26)
[ERROR] [build] Caught error rendering /og.png: Error: Cannot find the font path.
```

**Additional context**

After adding diagnostic logs in `og.png.ts`, `fontData["--font-inter"]` contains 8 entries instead of 4: each weight (400, 500, 600, 700) appears twice. One entry per weight has a `src` array with only `woff2`, the other has `src` with both `woff` and `truetype`.

The current implementation in `src/utils/getFontPathByWeight.ts`:

```typescript
return fonts
  .find(font => font.weight === String(weight) && font.style === style)
  ?.src.find(file => file.format === format)?.url;
```

The outer `.find()` returns the first weight match (the woff2-only entry), and the inner `.find()` searches for `truetype` within that entry, finds nothing, and returns undefined.

**Proposed fix**

Iterate through all entries matching the weight and return the URL only when both weight AND format match:

```typescript
export function getFontPathByWeight(
  fonts: FontData[],
  weight: number,
  options?: {
    style?: "normal" | "italic";
    format?: string;
  }
): string | undefined {
  const style = options?.style ?? "normal";
  const format = options?.format ?? "truetype";

  for (const font of fonts) {
    if (font.weight === String(weight) && font.style === style) {
      const src = font.src.find(file => file.format === format);
      if (src) return src.url;
    }
  }

  return undefined;
}
```

This handles cases where multiple font entries exist per weight while preserving the original behavior for fonts with a single entry per weight.

**Environment**

- AstroPaper version: 6.x (latest as of May 2026)
- Astro version: 5.x
- OS: Windows 11 (also reproduced on Netlify build running Linux)
- Node version: v24.11.1

**Willing to contribute**

I have a working fix and can submit a Pull Request if maintainers agree with the proposed approach.

--- a/src/utils/getFontPathByWeight.ts
+++ b/src/utils/getFontPathByWeight.ts
@@ -11,7 +11,12 @@ export function getFontPathByWeight(
   const style = options?.style ?? "normal";
   const format = options?.format ?? "truetype";
 
-  return fonts
-    .find(font => font.weight === String(weight) && font.style === style)
-    ?.src.find(file => file.format === format)?.url;
+  for (const font of fonts) {
+    if (font.weight === String(weight) && font.style === style) {
+      const src = font.src.find(file => file.format === format);
+      if (src) return src.url;
+    }
+  }
+
+  return undefined;
 }

--- a/src/utils/getFontPathByWeight.ts
+++ b/src/utils/getFontPathByWeight.ts
@@ -13,7 +13,7 @@ export function getFontPathByWeight(
 
   for (const font of fonts) {
     if (font.weight === String(weight) && font.style === style) {
-      const src = font.src.find(file => file.format === format);
+      const src = font.src.find(file => file.format === format) ?? font.src[0];
       if (src) return src.url;
     }
   }

